crates/corro-agent/src/api/peer.rs

```diff
@@ -702,6 +711,53 @@ fn handle_need(
                 }
             }
         }
+        SyncNeedV1::Empty { ts } => {
+            if last_cleared_ts.is_none() {
+                return Ok(());
+            }
+            let last_cleared_ts = last_cleared_ts.unwrap();
+            let mut filter = String::from("");
+            if let Some(ts) = ts {
+                if ts >= last_cleared_ts {
+                    warn!("received empties timestamp {ts:?} greater our last cleared ts: {last_cleared_ts:?}");
+                    return Ok(());
+                }
+
+                filter = format!("ts > {ts}");
+            }
```

Comment:
I think `filter` could be avoided here. it doesn't matter much, but you could just trust the `ts` supplied by the requesting node and always filter like `ts > {ts}`. worse case: no results. I think the logic would be simpler. `ts` could default to `0`.